example/provider-local/garden/local/kustomization.yaml

```diff
@@ -2,8 +2,8 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
 resources:
-- controllerregistration.yaml
 - controllerdeployment.yaml
+- controllerregistration.yaml
```

Comment:
Why was this changed?

Reply:
Same reason as https://github.com/gardener/gardener/pull/10239#discussion_r1703953903

Reply:
Fair enough, but why was this working before/is needed now (with this PR)?

Reply:
The behaviour changed when I added the ordering to https://github.com/gardener/gardener/blob/0f11fd7cd17542cb6f258cd87447c4b02ed50552/example/provider-local/garden/skaffold/kustomization.yaml#L4-L5

I guess that these settings are inherited and propagated to referenced kustomization files.
https://github.com/gardener/gardener/blob/0f11fd7cd17542cb6f258cd87447c4b02ed50552/example/provider-local/garden/skaffold/kustomization.yaml#L7-L9